packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -36,120 +33,33 @@ export default function defaultResolver(
 
   const resolve = options.browser ? browserResolve : resolveSync;
 
-  let result = resolve(path, {
+  const result = resolve(path, {
     basedir: options.basedir,
-    defaultResolver,
     extensions: options.extensions,
+    isDirectory,
+    isFile,
     moduleDirectory: options.moduleDirectory,
     paths: options.paths,
-    rootDir: options.rootDir,
+    preserveSymlinks: false,
```

Comment:
Can we pass the node flag (if provided) here?

Reply:
we don't sniff out node flags anywhere else besides https://github.com/facebook/jest/blob/c4ae594d1f09fe37542da0597ad979dacdf27345/packages/jest-worker/src/workers/NodeThreadsWorker.ts#L71

We could start doing that I guess? This will keep the current behaviour for now (and `resolve` will default to `false` in the next major).

Reply:
we also do `realpath` right after, so I'd rather stick to current behavior and revisit later. Unless you've got a nice, clean solution here?

Reply:
I'm good with realpath and happy we don't branch anymore. Would need to spend more time playing with preserving symlinks, which I wish I had 😅